Branch for merges with upstream

LNx2 opened the pull request:
Fixed some more bugs, and did some clean up in `./scripts_src/global/gl_k_alcohl.ssl`  so that the alcohol addiction works better, ref #298

burner1024 commented on scripts_src/global/gl_k_alcohl.ssl:
```diff
@@ -2,18 +2,18 @@
-#define TIME_TWO_DAYS 1728000
-#define TIME_THREE_DAYS 2592000
-#define TIME_ONE_WEEK 6048000
+#define SHAKES_TIME (2*ONE_GAME_DAY)
```
Let's keep define names starting with classificator (`TIME_x`, etc).

burner1024 commented on scripts_src/global/gl_k_alcohl.ssl:
```diff
@@ -62,24 +70,38 @@ procedure drink(variable who, variable item) begin
-    else chance = 10;
+    if ((game_time - last_drink_date) >= SPORADIC_ALCOHOL_USAGE_TIME) then begin
+        chance = 1;
+//        display_msg("<test> last drink longer than a week ago. Your base chance of addiction is : "+chance);
```
Don't need debug leftovers, here or anywhere else.

burner1024 commented on scripts_src/global/gl_k_alcohl.ssl:
```diff
@@ -62,24 +70,38 @@ procedure drink(variable who, variable item) begin
     if has_trait(TRAIT_TRAIT, dude_obj, TRAIT_drug_addict) then chance = chance * 2;
     // Drug resistant: half chance
     if has_trait(TRAIT_TRAIT, dude_obj, TRAIT_drug_resistant) then chance = chance * 0.5;
+    if is_booze(item) then chance = chance * 2;
```
I don't think strong alcohol is more addictive, actually, it's more the regularity of it.

LNx2 replied:
Well without this change drinking two beers directly one after the other has a way higher addiction chance than drinking one bottle of booze, even though that bottle of booze has more alcohol in it and the time of consumption is more or less the same.
(Consuming two beers has almost a 6% chance to get addicted, consuming one booze has just 1%)

But I really don't feel all that strongly about it, I'll just revert the patch then. 🤷

burner1024 replied:
I think good approximation would be arctangent function, on both daily and weekly level, but like I said, too lazy to implement it properly. For now can just leave it like that, maybe add extra features later.

burner1024 commented on scripts_src/global/gl_k_alcohl.ssl:
```diff
@@ -136,12 +154,15 @@ procedure start begin
         set_critter_stat(dude_obj, STAT_ag, 1);
         set_critter_stat(dude_obj, STAT_ch, 1);
         display_mymsg(MSG_relieved);
+        display_msg(string_format (message_str_game(GAME_MSG_ITEM, 1), 1, message_str_game(GAME_MSG_STAT, 105) ) ); // "You gained 1 Agility"
```
Think won't work at all. You presume English, while in other languages phrase structure can be completely different.
Additionally, separate messages seem unnecessary and just pollute PDA screen. I would combine then.
Also, for clarity I would say "regained", rather than "gained".

LNx2 replied:
Fair enough, I just tried to re-use already existing strings, since I reasoned, I at least wouldn't introduce new problems. I guess there will be similar problems once someone tries to translate the strings used by the engine directly, but no need to solve everything at once.

I just put the strings into the msg file then.

LNx2 commented:
@burner1024 you can take another look now, and see if you are happy with the changes.

burner1024 commented on scripts_src/global/gl_k_alcohl.ssl:
```diff
@@ -70,34 +60,26 @@ procedure drink(variable who, variable item) begin
 
     // regular drinking increases chances
-    if ((game_time - last_drink_date) >= SPORADIC_ALCOHOL_USAGE_TIME) then begin
+    if ((game_time - last_drink_date) >= ONE_GAME_WEEK) then begin
```
No, I didn't mean to remove time defines, just rename them like `TIME_ALCOHOL_USAGE_SPORADIC`.
Because that's what you usually expect in completion.

burner1024 commented on data/text/english/game/k_alcohl.msg:
```diff
@@ -1,4 +1,4 @@
 {2}{}{The cravings are getting worse.}
 {3}{}{You've got the shakes. You have this intense urge to drink something with alcohol in it.}
-{4}{}{You feel as if a great burden has been lifted from your mind. No longer do you crave the need to drink alcohol.}
+{4}{}{You feel as if a great burden has been lifted from your mind. No longer do you crave the need to drink alcohol. You regained 1 Agility and 1 Charisma}
```
Please check master changes, I updated the line.

LNx2 commented:
@burner1024 you can take another look again.

> Please check master changes, I updated the line.
That patch was on the fix/cassidy-voiceline branch, that's why I didn't see it. I rebased your changes to data/text/english/game/k_alcohl.msg onto the master branch commit "0c05481d0" and then merged them into the rest of my changes, I hope that doesn't cause too much confusion in the git history.

burner1024 commented:
My bad, pushed to the wrong branch.
Thanks.
